src/Cube.hs

```diff
@@ -0,0 +1,31 @@
+module Cube where
+
+import qualified Data.Vector.Unboxed as V
+
+data Corner = Corner (V.Vector Int) (V.Vector Int) deriving (Show)
+data Edge = Edge (V.Vector Int) (V.Vector Int) deriving (Show)
+data Cube = Cube { corners :: Corner, edges :: Edge } deriving (Show)
+
+class Orientation o where
+    orien :: o -> V.Vector Int
+class Permutation p where
+    perm :: p -> V.Vector Int
```

Comment:
@SaxJ these could probably be a single type class; the only reason I didn't do it was because I couldn't think of a good name. Do you have any opinion or suggestion?